File: backend/app/schemas/result.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class ResultRead(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    job_id: uuid.UUID
    module_name: str
    minio_key: str | None
    summary: dict
    created_at: datetime


class ResultDownloadURL(BaseModel):
    url: str
    expires_in: int = 3600

File: frontend/src/types/result.ts
export interface Result {
  id: string;
  job_id: string;
  module_name: string;
  minio_key: string | null;
  summary: Record<string, unknown>;
  created_at: string;
}

export interface DownloadURL {
  url: string;
  expires_in: number;
}
